api/dbadapters/index.ts

```diff
@@ -66,3 +68,74 @@ register("postgres", RedshiftDbAdapter);
 register("redshift", RedshiftDbAdapter);
 register("snowflake", SnowflakeDbAdapter);
 register("sqldatawarehouse", SQLDataWarehouseDBAdapter);
+
+export function constructEvaluationFromQueryOrAction(
+  queryOrAction: string | dataform.Table | dataform.Operation | dataform.Assertion,
+  concatenate: boolean,
+  queryModifier?: (mod: string) => string
+) {
+  if (!queryModifier) {
+    queryModifier = (query: string) => query;
+  }
+  const validationQueries = new Array<dataform.IValidationQuery>();
+  if (typeof queryOrAction === "string") {
+    validationQueries.push({ query: queryModifier(queryOrAction) });
+  } else {
+    try {
+      if (queryOrAction instanceof dataform.Table) {
+        if (queryOrAction.type === "incremental") {
+          const incrementalTableQueries = queryOrAction.incrementalPreOps.concat(
+            queryOrAction.incrementalQuery,
+            queryOrAction.incrementalPostOps
+          );
+          if (concatenate) {
+            validationQueries.push({
+              query: concatenateQueries(incrementalTableQueries, queryModifier),
+              incremental: true
+            });
+          } else {
+            incrementalTableQueries.forEach(query =>
+              validationQueries.push({ query: queryModifier(query), incremental: true })
+            );
+          }
+        }
+        const tableQueries = queryOrAction.preOps.concat(
+          queryOrAction.query,
+          queryOrAction.postOps
+        );
+        if (concatenate) {
+          validationQueries.push({
+            query: concatenateQueries(tableQueries, queryModifier)
+          });
+        } else {
+          tableQueries.forEach(query => validationQueries.push({ query: queryModifier(query) }));
+        }
+      } else if (queryOrAction instanceof dataform.Operation) {
+        // TODO: The prefix method is a bit sketchy for operations. For example, after attaching the `eplain` prefix
```

Comment:
This issue already exists with the current method of evaluating operations, so is not a regression

Reply:
does this not happen for all action types, though? e.g. if i have a `table` action that looks like this:

```
// some comment
select 1
```